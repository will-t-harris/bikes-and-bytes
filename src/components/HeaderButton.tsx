import React from "react"
import { Link } from "gatsby"

interface Props {
  buttonTitle: string
  buttonSlug: string
  buttonClass: string
}

const HeaderButton = ({ buttonTitle, buttonSlug, buttonClass }: Props) => (
  <>
    <div className="flex flex-auto">
      <Link
        to={buttonSlug}
        className={buttonClass}
        activeClassName="text-pink-600 underline transition-all ease-in duration-200"
      >
        {buttonTitle}
      </Link>
    </div>
  </>
)

export default HeaderButton
