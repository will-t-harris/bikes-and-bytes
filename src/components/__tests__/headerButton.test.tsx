import React from "react"
import { render } from "@testing-library/react"

import HeaderButton from "../HeaderButton"

const headerButtonProps = {
  buttonTitle: "Test title",
  buttonSlug: "/testSlug",
  buttonClass: "mx-auto",
}

describe("Header Button", () => {
  it("matches the previous snapshot", () => {
    const { container } = render(<HeaderButton {...headerButtonProps} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the button title", () => {
    const { getByText } = render(<HeaderButton {...headerButtonProps} />)

    expect(getByText("Test title")).toBeInTheDocument()
  })
})
