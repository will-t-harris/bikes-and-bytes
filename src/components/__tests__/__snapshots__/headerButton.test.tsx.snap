// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Button matches the previous snapshot 1`] = `
<a
  class="mx-auto"
  href="/testSlug"
>
  Test title
</a>
`;
